querydsl-mongodb/src/main/java/com/querydsl/mongodb/MongodbSerializer.java

```diff
@@ -127,6 +127,11 @@ public Object visit(Operation<?> expr, Void context) {
             if (subOp == Ops.IN) {
                 return visit(OperationImpl.create(Boolean.class, Ops.NOT_IN, subOperation.getArg(0),
                         subOperation.getArg(1)), context);
+            } else if (subOp == Ops.COL_IS_EMPTY) {
```

Comment:
Do you mean instead of relying on the implementation of `COL_IS_EMPTY` (which uses $or) detecting $or and change that to $nor?
How deep do you want to detect?
If there is an AND involved, with two ORs which have two leaves (`x.isEmpty().and(y.isEmpty()).not()` it could cause problems.
Does Inverting the $or then work?